cli/internal/terraform/loader.go

```diff
@@ -39,27 +42,29 @@ func prepareWorkspace(fileHandler file.Handler, provider cloudprovider.Provider)
 		if err != nil {
 			return err
 		}
-		fileName := strings.TrimPrefix(path, rootDir+"/")
-		return fileHandler.Write(fileName, content, file.OptMkdirAll)
-	})
-}
-
-// cleanUpWorkspace removes files that were loaded into the workspace.
-func cleanUpWorkspace(fileHandler file.Handler) error {
-	// try to remove any terraform files in the workspace
-	for _, csp := range []string{"aws", "azure", "gcp", "qemu"} {
-		rootDir := path.Join("terraform", csp)
-		if err := fs.WalkDir(terraformFS, rootDir, func(path string, d fs.DirEntry, err error) error {
+		fileName := strings.Replace(filepath.Join(workingDir, path), rootDir+"/", "", 1)
+		if err := fileHandler.Write(fileName, content, file.OptMkdirAll); errors.Is(err, afero.ErrFileExists) {
+			// If a file already exists, check if it is identical. If yes, continue and don't overwrite.
+			// If no, don't overwrite it but throw an error. It could be from a different version of Constellation.
+			existingFileContent, err := fileHandler.Read(fileName)
 			if err != nil {
 				return err
 			}
-			fileName := strings.TrimPrefix(path, rootDir+"/")
-			return ignoreFileNotFoundErr(fileHandler.RemoveAll(fileName))
-		}); err != nil {
+
+			if !bytes.Equal(content, existingFileContent) {
+				return fmt.Errorf("trying to overwrite existing Terraform file with different version")
+			}
```

Comment:
I feel like reading both files and comparing their byte content isnt the ideal way to check for equality, but should be OK for this case, since this code is only used by the CLi

Reply:
Mh, not sure to be honest. Initially I had the idea to compare SHA256 hashes, but that essentially ends up also being a byte-by-byte comparison.

I think the main issue here is that both files are loaded into memory. However, luckily Terraform files aren't this large so it shouldn't be a major issue I guess.

Reply:
@thomasten Any suggestion on how to do file equality comparisons here? Is bytes.Equal over the whole content fine (they are not large files) versus using a hash or similar?

Reply:
Generally, prefer simple implementations over (prematurely) optimized, but more complex ones. (See our review guide.) So IMO this is the right implementation in this case.
If you would write a generic file comparison func, you would read the files in chunks. I think you would only use hashes in special case, e.g., if you have a large file that you want to compare against multiple others. But even in this case, this isn't always the best solution because you need to read all the other files completely even if they already differ in the first chunk.